src/promptflow/promptflow/executor/_tool_invoker.py

```diff
@@ -2,9 +2,99 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
+import docutils.nodes
+from dataclasses import dataclass
+from docutils.core import publish_doctree
+from functools import partial
+from typing import Callable
+
 from promptflow._core.tool import ToolInvoker
+from promptflow.contracts.flow import InputAssignment, Node, ToolSource
+from promptflow.executor._tool_resolver import ToolResolver
 
 
 class DefaultToolInvoker(ToolInvoker):
     def invoke_tool(self, f, *args, **kwargs):
         return f(*args, **kwargs)  # Do nothing
+
+
+@dataclass
+class AssistantTool():
+    name: str
+    description: dict
+    func: Callable
+
+
+class AssistantToolInvoker():
+    def __init__(self):
+        self._assistant_tools = {}
+
+    def load_tools(self, tools: list):
+        tool_resolver = ToolResolver.active_instance()
+        for tool in tools:
+            if tool["type"] != "function":
+                self._assistant_tools[tool["type"]] = AssistantTool(name=tool["type"], description=tool, func=None)
+                continue
+            inputs = tool.get("predefined_inputs", {})
```

Comment:
Have better names for inputs and updated_inputs

Reply:
Updated.